packages/altair-app/src/app/modules/altair/effects/query-collection.effect.ts

```diff
@@ -155,24 +155,34 @@ export class QueryCollectionEffects {
   }, { dispatch: false });
 
 
-  importCollection$: Observable<Action> = createEffect(() => {
-    return this.actions$
-      .pipe(
-        ofType(collectionActions.IMPORT_COLLECTION),
-        switchMap(() => {
-          return from(openFile({ accept: '.agc' }));
-        }),
-        switchMap((data: string) => this.collectionService.importCollectionDataFromJson(data)),
-        tap(() => this.notifyService.success('Successfully imported collection.')),
-        map(() => new collectionActions.LoadCollectionsAction()),
-        catchError((error) => {
-          const errorMessage = error.message ? error.message : error.toString();
-          this.notifyService.error(`Something went wrong importing collection. Error: ${errorMessage}`);
-          return EMPTY;
-        }),
-      )
+
+  importCollections$: Observable<Action> = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(collectionActions.IMPORT_COLLECTIONS),
+      switchMap(() => {
+        return from(openFiles({ accept: ".agc" }));
+      }),
+      pipe(filter<string[]>((e) => e !== undefined)),
```

Comment:
Is the `pipe()` needed here, since you are already streaming observables? Also not sure why we need to filter `undefined` here. If I understand this correctly, `e` would be an array of strings, and since we use `Promise.all` in `openFiles`, we should always get an array. Or are you trying to filter out `undefined` from the elements in the array? If that's the case, I would recommend filtering in `openFiles` itself, otherwise filter it before using it within the function below, because this `pipe(filter(...))` wouldn't do that.